File: src/main/java/com/googongill/aditory/controller/dto/category/response/ImportCategoryResponse.java
package com.googongill.aditory.controller.dto.category.response;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.googongill.aditory.domain.Category;
import com.googongill.aditory.service.dto.category.CategoryResult;
import lombok.Builder;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

@Builder
@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class ImportCategoryResponse {
    @Builder.Default
    private List<CategoryResult> userCategories = new ArrayList<>();

    public static ImportCategoryResponse of(List<Category> categories) {
        return ImportCategoryResponse.builder()
                .userCategories(categories.stream()
                        .map(category -> CategoryResult.builder()
                                .categoryId(category.getId())
                                .categoryName(category.getCategoryName())
                                .build())
                        .collect(Collectors.toList())
                )
                .build();
    }
}

File: src/main/java/com/googongill/aditory/controller/dto/category/response/ImportCategoryResponse.kt
package com.googongill.aditory.controller.dto.category.response

import com.fasterxml.jackson.annotation.JsonAutoDetect
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.googongill.aditory.domain.Category
import com.googongill.aditory.service.dto.category.CategoryResult

@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
data class ImportCategoryResponse(
    val userCategories: List<CategoryResult> = emptyList()
) {

    companion object {
        fun of(categories: List<Category>): ImportCategoryResponse {
            return ImportCategoryResponse(
                userCategories = categories.map {category ->
                    CategoryResult(
                        categoryId = category.id,
                        categoryName = category.categoryName
                    )
                }
            )
        }
    }

}
